src/BoundaryConditions/perturbation_advection_open_boundary_matching_scheme.jl

```diff
@@ -101,14 +103,14 @@ end
     Δt = clock.last_stage_Δt
     Δt = ifelse(isinf(Δt), 0, Δt)
 
-    ūⁿ⁺¹ = getbc(bc, l, m, grid, clock, model_fields)
-    uᵢⁿ     = @inbounds getindex(u, boundary_secret_storage_indices...)
+    ūⁿ⁺¹    = getbc(bc, l, m, grid, clock, model_fields)
+    uᵢⁿ     = ifelse(clock.iteration == 0, ūⁿ⁺¹, @inbounds getindex(u, boundary_secret_storage_indices...))
```

Comment:
@jagoosw this is the fix for the initial condition. It simply overwrites the first iteration to get the external BC, since in the first iteration the secret storage isn't set yet.

Reply:
I see that makes sense!

Reply:
Should you be using regularize boundary condition to obtain this information?

Reply:
I've seen these kinds of designs go badly in other context (mostly with ClimaAtmos) so I think we should endeavor to use a different design if at all possible, which does not require a running simulation for initialization

Reply:
I tested a few different configurations, and for every script I tested, it made no difference if we used a "secret storage" for the left boundary or not. That is, things work equally well whether we treat the left boundary differently from the right boundary, or treat both of them the same.

So with that information, we may also just remove this asymmetry between sides, simplify things, and treat the left boundary the same as the right.

Reply:
i was just referring to the concept of something being "secret"

Reply:
I changed the name to `previous_timestep_storage_indices`, which I believe to be more accurate.

Reply:
What are the indices used for?

Reply:
They point to the location in `model.velocities.u` (and v, w) that store the previous time step. In particular @jagoosw uses `i=0`.